apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/AppAndServiceRegisterClient.java

```diff
@@ -48,13 +52,14 @@
 @DefaultImplementor
 public class AppAndServiceRegisterClient implements BootService, GRPCChannelListener, Runnable, TracingContextListener {
     private static final ILog logger = LogManager.getLogger(AppAndServiceRegisterClient.class);
-    private static final String PROCESS_UUID = UUID.randomUUID().toString().replaceAll("-", "");
+    private static final String INSTANCE_UUID = Config.Agent.INSTANCE_UUID.equals("") ? UUID.randomUUID().toString().replaceAll("-", "") : Config.Agent.INSTANCE_UUID;
```

Comment:
Please set UUID the first it is used, don't init here to avoid `Config.Agent.INSTANCE_UUID` doesn't init in time. In the old day, we didn't need to worry about this. So please move the initialization codes.

Reply:
Don't use `.equals()`, use `StringUtil#isEmpty`